src/components/search/SearchBar.tsx

```diff
@@ -0,0 +1,180 @@
+import * as React from 'react';
+import styles from './Search.module.scss';
+// @ts-ignore
+import { unnnicInput } from '@weni/unnnic-system';
+import { applyVueInReact } from 'vuereact-combined';
+import AppState from 'store/state';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import {
+  DispatchWithState,
+  HandleSearchChange,
+  handleSearchChange,
+  onUpdateCanvasPositions
+} from 'store/thunks';
+import { RenderNodeMap, Search } from 'store/flowContext';
+import i18n from 'config/i18n';
+import Button, { ButtonTypes } from 'components/button/Button';
+
+const UnnnicInput = applyVueInReact(unnnicInput, {
+  vue: {
+    componentWrapAttrs: {
+      'unnnic-input': 'true'
+    }
+  }
+});
+export interface SearchStoreProps {
+  search?: Search;
+  nodes?: RenderNodeMap;
+  handleSearchChange?: HandleSearchChange;
+}
+
+export class SearchBar extends React.PureComponent<SearchStoreProps, {}> {
+  private handleInput(value: string) {
+    const nodes = this.findNodes(value);
+    this.props.handleSearchChange({
+      active: true,
+      value: value,
+      nodes: nodes,
+      selected: 0
+    });
+    this.dragBackground();
+  }
+
+  private getAllNodes() {
+    return Object.entries(this.props.nodes).map(item => {
+      return {
+        uuid: item[0],
+        data: item[1]
+      };
+    });
+  }
+  private findNodes(value: string) {
+    const nodes = this.getAllNodes();
+    return nodes.filter(item => {
+      const nodeItem: any = item.data.node.actions;
+      if (nodeItem.length > 0) {
+        //find by type
+        const foundTitle = i18n
+          .t(`actions.${nodeItem[0].type}.name`)
+          .toLocaleLowerCase()
+          .includes(value.toLocaleLowerCase());
+        //find by content
+        const foundContent = nodeItem[0].hasOwnProperty('text')
+          ? item.data.node.actions[0].text.includes(value)
+          : nodeItem[0].hasOwnProperty('result_name')
+          ? item.data.node.actions[0].result_name.includes(value)
+          : false;
+
+        return foundTitle || foundContent;
+      } else {
+        return false;
+      }
+    });
+  }
+
+  private dragBackground() {
+    const canvasBg = document.getElementById('panzoom');
+    if (this.props.search.nodes[this.props.search.selected]) {
+      const ui = this.props.search.nodes[this.props.search.selected].data.ui.position;
```

Comment:
Destructuring `this.props.search` will result in fewer repetitions, which in addition to making the code more readable, will make maintenance easier:

```suggestion
    const { nodes, selected } = this.props.search
    if (nodes[selected]) {
      const ui = nodes[selected].data.ui.position;
```